application/ingestr/kafka-demo.xsh

```diff
@@ -0,0 +1,189 @@
+        title "Deleting Kafka topic"
+        docker compose --file $COMPOSE_FILE run --rm --no-TTY delete-topic
+        echo "Done."
+        return self
+
+
+class Datawrapper:
+    """
+    Wrap actions around data:
+
+    - Acquire data from public resource.
+    - Publish data to Kafka topic.
+    - Load Kafka topic into CrateDB table.
+    - Verify data in CrateDB table.
+    """
+
+    def acquire(self):
+
+        if p"nyc-yellow-taxi-2017-subset.ndjson".exists():
+            return self
+
+        title "Acquiring NDJSON data"
+
+        # Acquire NYC Taxi 2017 dataset in JSON format (~90 MB).
+        wget --no-clobber --continue "https://gist.githubusercontent.com/kovrus/328ba1b041dfbd89e55967291ba6e074/raw/7818724cb64a5d283db7f815737c9e198a22bee4/nyc-yellow-taxi-2017.tar.gz"
+
+        # Extract archive.
+        tar -xvf nyc-yellow-taxi-2017.tar.gz
+
+        # Create a subset of the data (5000 records) for concluding the first steps.
+        cat nyc-yellow-taxi-2017.json | head -n 5000 > nyc-yellow-taxi-2017-subset.ndjson
+
+        echo "Done."
+        return self
+
+    def publish(self):
+
+        # Publish data to the Kafka topic.
+        title "Publishing NDJSON data to Kafka topic"
+        cat nyc-yellow-taxi-2017-subset.ndjson | docker compose --file $COMPOSE_FILE run --rm --no-TTY publish-topic
+        echo "Done."
+
+        # Wait a bit for the data to transfer and converge successfully.
+        sleep 3
+
+        return self
+
+    def load(self):
+
+        # Invoke ingestr job.
+        title "Invoking ingestr job"
+
+        uvx --python=3.12 --prerelease=allow --with-requirements=requirements.txt \
+            ingestr ingest --yes \
+            --source-uri "kafka://?bootstrap_servers=localhost:9092&group_id=test_group&value_type=json&select=value" \
+            --source-table "demo" \
+            --dest-uri "cratedb://crate:crate@localhost:5432/?sslmode=disable" \
+            --dest-table "doc.kafka_demo"
+
+        echo "Done."
+        return self
+
+    def display(self):
+
+        title "Displaying data in CrateDB"
+
+        docker compose --file $COMPOSE_FILE run --rm httpie \
+            http "$CRATEDB_HTTP_URL/_sql?pretty" 'stmt=REFRESH TABLE "kafka_demo";' --ignore-stdin > /dev/null
+
+        docker compose --file $COMPOSE_FILE run --rm httpie \
+            http "$CRATEDB_HTTP_URL/_sql?pretty" 'stmt=SELECT * FROM "kafka_demo" LIMIT 5;' --ignore-stdin
+
+        docker compose --file $COMPOSE_FILE run --rm httpie \
+            http "$CRATEDB_HTTP_URL/_sql?pretty" 'stmt=SELECT COUNT(*) FROM "kafka_demo";' --ignore-stdin
+
+        return self
+
+    def verify(self):
+        title "Verifying data in CrateDB"
+        $size_reference=5000
+        $size_actual=$(
+            docker compose --file $COMPOSE_FILE run --rm httpie \
+                http "$CRATEDB_HTTP_URL/_sql?pretty" 'stmt=SELECT COUNT(*) FROM "kafka_demo";' --ignore-stdin \
+                | jq .rows[0][0]
+        )
```

Comment:
This command still needs `jq` to be installed on your workstation. Maybe it's better to use `crash` instead?
Probably, it's better to use Python stdlib's built-in HTTP client and the JSON module natively, to further reduce noise here.

Reply:
9e82a835626 makes it much better, no longer relying on the container invocation, but doing it with pure Python instead.